File: 190.reverse_bits/reverseBits.py
class Solution:
# @param n, an integer
# @return an integer
    def reverseBits(self, n):
        res = 0
        for i in range(32):
            res += 1 & n
            if i == 31:
                return res
            n >>= 1
            res <<= 1

s = Solution()
print s.reverseBits(43261596)

File: 190.reverse_bits/reverseBits.js
/**
 * @param {number} n - a positive integer
 * @return {number} - a positive integer
 */
var reverseBits = function(n) {
    if (n == 0) return 0;

    var result = 0;
    for (var i = 0; i < 32; i++) {
        result <<= 1;
        if ((n & 1) == 1) result++;
        n >>= 1;
    }
    return result;
};

console.log(reverseBits(1));